students/yushus/session06/mailroom4.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env/python3
+
+"""
+Yushu Song
+Au2018-Py210B
+Mailroom4 assignment
+"""
+
+import os
+import random
+import re
+import sys
+
+donor_db = {"William Gates, III": [653772.32, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+            "Elon Musk": [250000, 15000]}
+
+prompt = "\n".join(("Welcome to mailroom bot!",
+          "Please choose from below options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Send letters to all donors",
+          "4 - quit",
+          ">>> "))
+
+def get_a_thank_you_letter(name, amount):
+    letter = f'Dear {name},\n\n\tThank you for your generous donation of ${amount}.\n\n\tYour kindness is really making the world different!\n\n\t\tSincerely,\n\t\tMailroom Bot'
+    return letter
+
+def update_donation(name, amount, donor_db):
+
+    # Each donation has to be non-negative amount and less than MAX value of the system allowed
+    if amount <= sys.maxsize and amount > 0:
+        donor_db[name].append(amount)
+        return True
+    else:
+        raise ValueError('Donation has to be more than 0')
+
+# Add donor if not exist
+def update_donor(name, donor_db):
+
+    # Check if name is null or empty
+    if not name or name.isspace() or name is None:
```

Comment:
Minor: `name is None` is included in `not name`, so `name is None` is not necessary.